### PR TITLE
Not to consider VPA recommendation if status conditions has type "ConfigUnsupported"

### PR DESCRIPTION
**Pull request by ggaurav10**
Not to consider VPA recommendation if status conditions has type "ConfigUnsupported"

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
Now HVPA doesn't take VPA recommendations into account if VPA condition has `ConfigUnsupported`, `ConfigDeprecated` or `LowConfidence` set to `true`
```

### CODE REVIEW

File: controllers/hvpa_controller.go [hunk 1]
```diff
@@ -844,6 +844,13 @@ func getWeightedRequests(vpaStatus *vpa_api.VerticalPodAutoscalerStatus, hvpa *a
 		return nil, false, nil, nil
 	}
 	for k, v := range vpaStatus.Conditions {
+		if v.Type == vpa_api.ConfigUnsupported || v.Type == vpa_api.ConfigDeprecated || v.Type == vpa_api.LowConfidence {
```
Comment (amshuman-kr) [inline controllers/hvpa_controller.go:847]: A suggestion for readability. Why not put the condition types in an array and loop over them? The code would be more easier to understand and extend in case more conditions are applicable later, WDYT?

**Review by amshuman-kr — CHANGES_REQUESTED** [review]
Apart from the comment below (which is not hard requirement), could we add a few unit tests to cover the additional cases introduced by this PR?

File: controllers/hvpa_controller.go (hunk 1 above)
Comment (ggaurav10) [reply controllers/hvpa_controller.go:847]: sure. sounds good. will do.
thanks.

Comment (ggaurav10) [reply controllers/hvpa_controller.go]: Done in https://github.com/gardener/hvpa-controller/pull/69/commits/8913e48fc948e79ce3ca0ba51e596cc915eb1ea6

**Review by amshuman-kr — CHANGES_REQUESTED** [review]
The changes look good. But how about the test case for the edge-cases introduced?

**Comment by ggaurav10** [comment]
added test cases in https://github.com/gardener/hvpa-controller/pull/69/commits/1b0b4c83455a65dd6ed4dcb779619a98c5ba4808

**Review by amshuman-kr — APPROVED** [review]
Thanks for the test case. LGTM
